### PR TITLE
[166] Generating: Test and UITest target

### PR DESCRIPTION
**Pull request by vnntsu**
https://github.com/nimblehq/ios-templates/issues/166

## What happened

Generating follow the structure:
```json
├── {ProjectName}Tests
│   ├── Configurations
│   │   └── Plists
│   │       └── Info.plist
│   ├── Resources
│   └── Sources
│       ├── Dummy
│       ├── Mocks
│       ├── Specs
│       └── Utilities
└── {ProjectName}UITests
    ├── Configurations
    │   └── Plists
    │       └── Info.plist
    ├── Resources
    └── Sources
```
 
## Insight

- Prepare some empty folder with .gitkeep inside
- Update `.gitignore` to ignore `Pods`, `Fastlane`'s output and `users`' folders
 
## Proof Of Work

Screen Shot 2021-09-07 at 10 30 00

### CODE REVIEW

**Review by llleyelll — COMMENTED** [review]
Should we move `UITests.swift` and `UnitTests.swift` to `Sources` folder?

**Comment by vnntsu** [comment]
> Should we move `UITests.swift` and `UnitTests.swift` to `Sources` folder?

Good catch, I think it should be removed now 😘

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,35 @@ Temporary Items
...
+# It is recommended to not store the screenshots in the git repo.
+# Instead, use fastlane to re-generate the screenshots whenever they are needed.
+# For more information about the recommended setup visit:
+# https://docs.fastlane.tools/best-practices/source-control/#source-control
+
+fastlane/README.md
+fastlane/report.xml
+fastlane/Preview.html
+fastlane/screenshots/**/*.png
+fastlane/test_output
```
Comment (llleyelll) [inline .gitignore:65]: Sorry, I missed this one 😓 . 
I think we should ignore generated files as well. What do you think?

```.gitignore
# Sourcery
*.generated.swift
```

File: Tuist/ProjectDescriptionHelpers/Target+Initializing.swift [hunk 2]
```diff
@@ -23,27 +25,31 @@ extension Target {
...
-            name: "\(name)Tests",
+            name: targetName,
             platform: .iOS,
             product: .unitTests,
-            bundleId: "\(bundleId).\(name)Tests",
-            infoPlist: .default,
-            sources: ["\(name)Tests/**"],
+            bundleId: "\(bundleId).\(targetName)",
+            infoPlist: "\(targetName)/\(plistsPath)/Info.plist",
+            sources: ["\(targetName)**"],
```
Comment (llleyelll) [inline Tuist/ProjectDescriptionHelpers/Target+Initializing.swift:35]: Sorry, I missed this one.
```suggestion
            sources: ["\(targetName)/**"],
```

File: Tuist/ProjectDescriptionHelpers/Target+Initializing.swift [hunk 3]
```diff
@@ -23,27 +25,31 @@ extension Target {
...
                 .target(name: name)
             ]
         )
     }
 
     public static func uiTestsTarget(name: String, bundleId: String = "co.nimblehq") -> Target {
+        let targetName = "\(name)UITests"
         return Target(
-            name: "\(name)UITests",
+            name: "\(targetName)",
```
Comment (llleyelll) [inline Tuist/ProjectDescriptionHelpers/Target+Initializing.swift:46]: Sorry I missed this one.
```suggestion
            name: targetName,
```

File: .gitignore (hunk 1 above)
Comment (vnntsu) [reply .gitignore:65]: @llleyelll  Sure :+1:

File: Tuist/ProjectDescriptionHelpers/Target+Initializing.swift [hunk 4]
```diff
@@ -23,27 +25,31 @@ extension Target {
...
+            name: targetName,
             platform: .iOS,
             product: .unitTests,
-            bundleId: "\(bundleId).\(name)Tests",
-            infoPlist: .default,
-            sources: ["\(name)Tests/**"],
+            bundleId: "\(bundleId).\(targetName)",
+            infoPlist: "\(targetName)/\(plistsPath)/Info.plist",
+            sources: ["\(targetName)/**"],
+            resources: ["\(targetName)/**/.gitkeep"], // To include empty folders
```
Comment (suho) [inline Tuist/ProjectDescriptionHelpers/Target+Initializing.swift:36]: Should we add `"\(targetName)/Resources/**" into this array?

File: Tuist/ProjectDescriptionHelpers/Target+Initializing.swift [hunk 5]
```diff
@@ -23,27 +25,31 @@ extension Target {
...
+            name: targetName,
             platform: .iOS,
             product: .uiTests,
-            bundleId: "\(bundleId).\(name)UITests",
-            infoPlist: .default,
-            sources: ["\(name)UITests/**"],
+            bundleId: "\(bundleId).\(targetName)",
+            infoPlist: "\(targetName)/\(plistsPath)/Info.plist",
+            sources: ["\(targetName)/**"],
+            resources: ["\(targetName)/**/.gitkeep"], // To include empty folders
```
Comment (suho) [inline Tuist/ProjectDescriptionHelpers/Target+Initializing.swift:52]: Should we add `"\(targetName)/Resources/**" into this array?

**Review by suho — APPROVED** [review]

**Pull request merged**